components/clarinet-deployments/src/types.rs

```diff
@@ -665,9 +837,69 @@ pub struct DeploymentSpecification {
     pub genesis: Option<GenesisSpecification>,
     pub plan: TransactionPlanSpecification,
     // Keep a cache of contract's (source, relative_path)
+    #[serde(with = "contracts_serde")]
     pub contracts: BTreeMap<QualifiedContractIdentifier, (String, FileLocation)>,
 }
 
+pub mod contracts_serde {
+    use bitcoincore_rpc::jsonrpc::base64::encode;
+    use clarinet_files::FileLocation;
+    use clarity_repl::clarity::vm::types::QualifiedContractIdentifier;
+    use serde::{ser::SerializeMap, Deserializer, Serializer};
+    use std::collections::{BTreeMap, HashMap};
+
+    use super::source_serde;
+
+    pub fn serialize<'ser, S>(
+        target: &'ser BTreeMap<QualifiedContractIdentifier, (String, FileLocation)>,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(target.len()))?;
+        for (k, v) in target {
+            let mut val = v.clone();
+            val.0 = encode(&v.0);
+
+            map.serialize_entry(&k.to_string(), &val)?;
+        }
+        map.end()
+    }
+
+    pub fn deserialize<'de, D>(
+        des: D,
+    ) -> Result<BTreeMap<QualifiedContractIdentifier, (String, FileLocation)>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let mut res: BTreeMap<QualifiedContractIdentifier, (String, FileLocation)> =
+            BTreeMap::new();
+        let container: HashMap<String, (String, HashMap<String, String>)> =
+            serde::Deserialize::deserialize(des)?;
+
+        for (qci_str, (src, file_location_map)) in container {
+            let qci = QualifiedContractIdentifier::parse(&qci_str).unwrap();
```

Comment:
Another unwrap we could replace with `.map_err(serde::de::Error::custom)?`.

Reply:
Done